VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -975,15 +966,19 @@ private boolean isScreenLandscape() {
         @Override
         public void run() {
             if (isResumed() && !isRemoving()) {
-                hideProgress();
-                if(player!=null) {
+                if (player != null) {
                     player.unfreeze();
+                    hideProgress();
                     if (player.isPlaying()) {
-                        hideProgress();
                         updateController("player unfreezed");
                     }
 
-                    if (pauseDueToDialog){
+                    if (IS_AUTOPLAY_ENABLED && isPrepared && !isAutoPlayDone) {
+                        isAutoPlayDone = true;
+                        player.start();
```

Comment:
I think the auto-play logic should be performed in `onPrepared()` instead of from the unfreeze callback.